weave-js/src/components/Panel2/PanelPlot/PanelPlot.tsx

```diff
@@ -531,20 +531,29 @@ const PanelPlotConfigInner: React.FC<PanelPlotProps> = props => {
       const fn = enabled
         ? TableState.enableGroupByCol
         : TableState.disableGroupByCol;
-      let newTable = fn(
+      let newTable = await fn(
         config.series[seriesIndex].table,
-        value
+        value,
+        inputNode,
+        weave,
+        stack
         // config.series[seriesIndex].dims[dimension.name as keyof SeriesConfig['dims']]
       );
       if (dimName === 'label') {
-        newTable = fn(newTable, config.series[seriesIndex].dims.color);
+        newTable = await fn(
+          newTable,
+          config.series[seriesIndex].dims.color,
+          inputNode,
+          weave,
+          stack
+        );
       }
       const newConfig = produce(config, draft => {
         draft.series[seriesIndex].table = newTable;
       });
       updateConfig(newConfig);
```

Comment:
Now that this is async, can you please condition this updateConfig call on mount status

Reply:
updateConfig is owned by caller, ignore